### PR TITLE
e2e: template resources

### PR DESCRIPTION
**Pull request by cbzzz**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This templates resource names (using the abbreviated commit hash) across the E2E tests to prevent naming collisions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [X] adds or updates e2e tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -138,7 +139,7 @@ test: manifests generate fmt vet envtest ## Run tests.
 
 .PHONY: e2etest
 e2etest: manifests generate local-deploy chainsaw
-	$(CHAINSAW) test ./e2e
+	echo "name: capl-e2e-$(GIT_REF)" | $(CHAINSAW) test ./e2e --values -
```
Comment (eljohnson92) [inline Makefile:142]: could this be supplied as just a flag rather than piping it to the command?

Comment (cbzzz) [reply Makefile:142]: No, we only have the following options: https://kyverno.github.io/chainsaw/latest/configuration/values/#example

Comment (eljohnson92) [reply Makefile:142]: I asked the chainsaw devs and they mentioned there is an `env()` function we could use to just read env variables in the tests, perhaps we could do that instead to just read in the GIT_REF?

Comment (cbzzz) [reply Makefile:142]: Unfortunately, Chainsaw seems to run tests in something like a subshell (e.g. `$USER` is accessible) so Make variables are not available.

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**